unix socket server

richardlehane opened the pull request:
I added an -fpr flag to sf v1.1 (but only when built with archivematica tag). This is a simple server that just implements the FPR interface (it only sends puid + error message) and runs on unix sockets - so faster than the http/json approach.

This python script is a client for the -fpr server. I've copied most of Misty's existing script for this with a slight change: on multiple matches this script will return an error rather than giving the first result (I feel this is safer bc when you get multiple matches - which only happens for results that are matched on extension only - they all have equal weight).

Hope this all makes sense!

richardlehane committed:
unix socket server

I added an -fpr flag to sf v1.1 (but only when built with archivematica tag). This is a simple server that just implements the FPR interface (it only sends puid + error message) and runs on unix sockets - so faster than the http/json approach.
 
This python script is a client for the -fpr server. I've copied most of Misty's existing script for this with a slight change: on multiple matches this script will return an error rather than giving the first result (I feel this is safer bc when you get multiple matches - which only happens for results that are matched on extension only - they all have equal weight).

Hope this all makes sense!

mistydemeo commented:
Wow, thanks very much for implementing this!

Is the server launched simply by doing `sf -fpr`? I've tried building using the "archivematica" tag on Mac and Linux, and the CLI flag doesn't seem to be recognized.

richardlehane commented:
Hi Misty, yep `sf -fpr` should launch it.
`go install -a -tags archivematica github.com/richardlehane/siegfried/cmd/sf` works on my linux. It won't work on Mac as the build tag mandates "archivematica" tag + linux: 
`// +build archivematica,linux` (in cmd/fpr.go)

Happy for this to change e.g. to +build archivematica,!windows. The corresponding tag in nofpr.go would also need to change e.g. +build archivematica windows

mistydemeo commented:
Oh, of course! The package was built using the `archivematica` config file, but not the `archivematica` tag, since the dh_golang buildtool we're using to build Debian packages still doesn't support go tags.

I wonder, how feasible do you think it'd be to make this a more generic feature that could be enabled in the default build? I'm wondering whether it's worth patching the source in our packages, or looking at making it a feature that could be enabled in the default build.

richardlehane commented:
Don't see any harm in that: have just pushed a commit putting fpr into the default build (excepting windows because unix sockets). Do you need another release?

mistydemeo commented:
I can patch around it for now - no need to rush a new release for me. Thanks though!

I'll take a look at this script tomorrow!

richardlehane committed:
Remove file call from siegfried fpr

Siegfried 1.2 includes a text matcher based on file's text detection (see the github.com/richardlehane/characterize package).
